File: lib/models/contacts.model.dart
class ContactsModel {
  int? id; // Proton @ ContactList.ContactID
  String serverContactID;
  String name;
  String email;
  String canonicalEmail;
  int isProton;

  ContactsModel({
    required this.id,
    required this.serverContactID,
    required this.name,
    required this.email,
    required this.canonicalEmail,
    required this.isProton,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'serverContactID': serverContactID,
      'name': name,
      'email': email,
      'canonicalEmail': canonicalEmail,
      'isProton': isProton,
    };
  }

  factory ContactsModel.fromMap(Map<String, dynamic> map) {
    return ContactsModel(
      id: map['id'],
      serverContactID: map['serverContactID'],
      name: map['name'],
      email: map['email'],
      canonicalEmail: map['canonicalEmail'],
      isProton: map['isProton'],
    );
  }
}

File: rust/src/proton_wallet/db/model/contacts_model.rs
use rusqlite::{Result, Row};
use serde::{Deserialize, Serialize};
use serde_rusqlite::from_row;

use super::model::ModelBase;

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ContactsModel {
    pub id: Option<u32>,
    pub server_contact_id: String,
    pub name: String,
    pub email: String,
    pub canonical_email: String,
    pub is_proton: u32,
}
impl ModelBase for ContactsModel {
    fn from_row(row: &Row) -> Result<Self> {
        Ok(from_row::<ContactsModel>(row).unwrap())
    }
}
